### ISSUE TITLE
API - Post photo not working any more

### ISSUE THREAD

**Issue by tcrouzet** [mgp25/Instagram-API#3016]
Since yesterday, all my posts stoped with "Invalid request options". I had the same problem 3 weeks ago and then suddenly everything went back to normal after 4 ou 5 days.

Same as issue #2972
https://github.com/mgp25/Instagram-API/issues/2972

Same problem on different accounts, with or without proxy.

**Comment by daptanque** [comment]
To me it started appearing such too, invalid request options, http 400, I tried some debuging the problem appears to show up on  mgp25/instagram-php/src/Client.php


        public function mapServerResponse(
        Response $responseObject,
        $rawResponse,
        HttpResponseInterface $httpResponse)
    {
        // Attempt to decode the raw JSON to an array.
        // Important: Special JSON decoder which handles 64-bit numbers!
        $jsonArray = $this->api_body_decode($rawResponse, true);

        // If the server response is not an array, it means that JSON decoding
        // failed or some other bad thing happened. So analyze the HTTP status
        // code (if available) to see what really happened.
        if (!is_array($jsonArray)) {
            $httpStatusCode = $httpResponse !== null ? $httpResponse->getStatusCode() : null;
            switch ($httpStatusCode) {
                case 400:
                    throw new \InstagramAPI\Exception\BadRequestException('Invalid request options. > ');


Anyway this for me is happening just on albums, I can still post photos and stories.

**Comment by damienmauchamp** [comment]
Same things here. This is what it returns when I'm trying to post a single photo in my timeline :

```
POST:  https://i.instagram.com/api/v1/upload/photo/
→ 203.82kB
← 400    58B
RESPONSE: Something is wrong. Please check your input and try again.

Something went wrong: Invalid request options.
```

It also started yesterday and I had the same issue few weeks ago just like OP said.

**Comment by daptanque** [comment]
Yeah, I tried to look up to Later to test the api but it's tricky they make a binding with the smartphone aplication and send command like if it was the smartphone doing the requests... Anyway I'm shifting my focus to the following links:

- https://www.instagram.com/developer/
- https://www.instagram.com/developer/changelog/
- https://developers.facebook.com/products/instagram/

**Comment by messilao** [comment]
I same problem.

**Comment by michelmelo** [comment]
I can post normally.
I am using latest version.

**Comment by daptanque** [comment]
@michelmelo album too?

**Comment by zhelinskyy** [comment]
Same problem!
video photo etc.

**Comment by michelmelo** [comment]
@daptanque  https://github.com/mgp25/Instagram-API/blob/master/examples/uploadAlbum.php

@zhelinskyy  https://github.com/mgp25/Instagram-API/blob/master/examples/uploadVideo.php
